src/rpp/rpp/operators/fwd/take_until.hpp

```diff
@@ -0,0 +1,81 @@
+//                   ReactivePlusPlus library
+//
+//           Copyright Aleksey Loginov 2022 - present.
+//                             TC Wang 2022 - present.
+//  Distributed under the Boost Software License, Version 1.0.
+//     (See accompanying file LICENSE_1_0.txt or copy at
+//           https://www.boost.org/LICENSE_1_0.txt)
+//
+//  Project home: https://github.com/victimsnino/ReactivePlusPlus
+
+#pragma once
+
+#include <rpp/observables/constraints.hpp>
+#include <rpp/observables/details/member_overload.hpp>
+#include <rpp/utils/function_traits.hpp>
+#include <rpp/utils/functors.hpp>
+
+namespace rpp::details
+{
+struct take_until_tag;
+}
+
+namespace rpp::details
+{
+
+template<constraint::decayed_type Type, constraint::observable TTriggerObservable>
+struct take_until_impl;
+
+template<constraint::decayed_type Type, typename SpecificObservable>
+struct member_overload<Type, SpecificObservable, take_until_tag>
+{
+
+    /**
+     * \brief Discard any items emitted by an Observable after a second Observable emits an item or terminates.
+     * \warning The take_until subscribes and begins mirroring the source Observable. It also monitors a second Observable that you provide. If this second Observable emits an item or sends a on_error/on_completed notification, the Observable returned by take_until stops mirroring the source Observable and terminates.
+     *
+     * \marble take_until
+       {
+           source observable        : +-1--2--3--|
+           until observable         : +--s--s----|
+           operator "take_until"    : +-1--------|
```

Comment:
Actually, it would be something like this
```suggestion
           operator "take_until"    : +-1|
```